daemon/daemon_test.go

```diff
@@ -1529,51 +1527,19 @@ func (s *daemonSuite) TestRequestContextCanceledOnStop(c *check.C) {
 			defer r.Body.Close()
 		}
 		c.Check(err, check.IsNil)
+		err = json.NewDecoder(r.Body).Decode(&resp)
+		c.Assert(err, check.IsNil)
 		close(clientC)
 	}()
 
 	<-gotReqC
 	d.Stop(nil)
-	reqErr := <-reqErrC
-	c.Check(errors.Is(reqErr, context.Canceled), check.Equals, true,
-		check.Commentf("unexpected error %v", reqErr))
 	<-clientC
```

Comment:
Something like this:

```suggestion
	select {
	case <-clientC:
		// request successfully cancelled
	case <-time.NewTimer(5 * time.Second).C:
		check.Error("request was not cancelled by the daemon stopping")
	}
```